docker_cli/image.py

```diff
@@ -0,0 +1,458 @@
+
+            .. warning::
+        This method does not work correctly if the image built does not have
+        bash installed.
+
+        Parameters
+        ----------
+        cmd : str
+            The desired command, in linux command format.
+        stdout : io.TextIOBase or subprocess special value or None, optional
+            The location to send the process stdout output to.Defaults to None.
+        stderr : io.TextIOBase or subprocess special value or None, optional
+            The location to send the process stderr output to.
+            Defaults to None.
+        interactive : bool, optional
+            A boolean describing whether or not to run this command in
+            interactive mode or not. Defaults to True.
+        network : str, optional
+            The name of the network. Defaults to "host".
+
+        Returns
+        -------
+        str, optional
+            The output of the process, if `stdout` == PIPE.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails.
+        CommandNotFoundOnImageError:
+            When a command is attempted that is not recognized on the image.
+        """
+        cmd = ["docker", "run", f"--network={network}", "--rm"]
+        if interactive:
+            cmd += ["-i"]
+            if stdin.isatty():
+                cmd += ["--tty"]
+        cmd += [self._id, "bash"]
+        cmd += ["-ci"] if interactive else ["-c"]
+        cmd += split(f"'{command}'")
+
+        try:
+            result = run(
+                cmd,
+                text=True,
+                stdout=stdout,  # type: ignore
+                stderr=stderr,  # type: ignore
+                check=True
+            )
+            retval = result.stdout
+        except CalledProcessError as err:
+            if err.returncode == 127:
+                raise CommandNotFoundOnImageError(err, split(command)[0])
+            else:
+                raise err
+        return retval
+
+    def check_command_availability(self, commands: Sequence[str]) -> List[str]:
+        """
+        Determines which of the commands in a list are present on the image.
+
+        Parameters
+        ----------
+        commands : Sequence[str]
+            A sequence of strings containing the names of commands to be
+            checked for.
+
+        Returns
+        -------
+        List[str]
+            The names of all commands in `commands` that were present on the
+            image.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails with a return value != 1.
+        """
+        found_commands = []
+        for command_name in commands:
+            try:
+                command = self.run(
+                    f"command -v {command_name}",
+                    stdout=PIPE
+                )
+                if command_name in command and \
```

Comment:
You've found what may be a relic of an earlier attempt to work the code, I think.

Typically `command -v [command]` seems to return a string that contains the path to the command, or it fails with code 1 if not. Initially I thought to just look for the name of the command in that return string (which is probably buggy in retrospect) but I might be able to just return True there instead, since the condition to return False assumes a CalledProcessError with value 1.